src/build_vehicle_gui.cpp

```diff
@@ -1354,6 +1416,9 @@ struct BuildVehicleWindow : Window {
 			/* Filter now! So num_engines and num_wagons is valid */
 			if (!FilterSingleEngine(eid)) continue;
 
+			/* Filter by name or NewGRF extra text */
+			if (!(FilterByName(e) || FilterByExtraText(e))) continue;
```

Comment:
This break the AND behavior when filtering with multiple words.
AND/OR and multiple words should be handled by `StringFilter`, so it is important to pass all data at once, and call `ResetState` and `GetState` only once.